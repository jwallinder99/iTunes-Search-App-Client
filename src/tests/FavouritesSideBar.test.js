import React from 'react';
import { render }  from '@testing-library/react';
import FavouriteSideBar from '../components/FavSideBar';

//dummy data
const mockFavourites = [
    {
        trackName: 'Sample Track 1',
        artistName: 'Sample Artist 1',
        artworkUrl100: 'sample_image_url_1',   
    },
    {
        trackName: 'Sample Track 2',
        artistName: 'Sample Artist 2',
        artworkUrl100: 'sample_image_url_2',      
    },
    {
        trackName: 'Sample Track 3',
        artistName: 'Sample Artist 3',
        artworkUrl100: 'sample_image_url_3',      
    },
]

test('Favourites Sidebar Component Snapshot', () => {
    const { asFragment } = render (
        <FavouriteSideBar favs={mockFavourites} removeFav={()=>{}}/>
    );
    expect( asFragment()).toMatchSnapshot()
})