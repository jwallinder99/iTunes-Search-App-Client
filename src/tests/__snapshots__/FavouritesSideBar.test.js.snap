// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favourites Sidebar Component Snapshot 1`] = `
<div>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-w60cv3-MuiButton-root{font-family:\\"Roboto\\",\\"Helvetica\\",\\"Arial\\",sans-serif;font-weight:500;font-size:0.875rem;line-height:1.75;letter-spacing:0.02857em;text-transform:uppercase;min-width:64px;padding:6px 8px;border-radius:4px;-webkit-transition:background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;color:#1976d2;}.css-w60cv3-MuiButton-root:hover{-webkit-text-decoration:none;text-decoration:none;background-color:rgba(25, 118, 210, 0.04);}@media (hover: none){.css-w60cv3-MuiButton-root:hover{background-color:transparent;}}.css-w60cv3-MuiButton-root.Mui-disabled{color:rgba(0, 0, 0, 0.26);}",
      }
    }
    data-emotion="css w60cv3-MuiButton-root"
  />
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-1e6y48t-MuiButtonBase-root-MuiButton-root{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;position:relative;box-sizing:border-box;-webkit-tap-highlight-color:transparent;background-color:transparent;outline:0;border:0;margin:0;border-radius:0;padding:0;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;vertical-align:middle;-moz-appearance:none;-webkit-appearance:none;-webkit-text-decoration:none;text-decoration:none;color:inherit;font-family:\\"Roboto\\",\\"Helvetica\\",\\"Arial\\",sans-serif;font-weight:500;font-size:0.875rem;line-height:1.75;letter-spacing:0.02857em;text-transform:uppercase;min-width:64px;padding:6px 8px;border-radius:4px;-webkit-transition:background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;color:#1976d2;}.css-1e6y48t-MuiButtonBase-root-MuiButton-root::-moz-focus-inner{border-style:none;}.css-1e6y48t-MuiButtonBase-root-MuiButton-root.Mui-disabled{pointer-events:none;cursor:default;}@media print{.css-1e6y48t-MuiButtonBase-root-MuiButton-root{-webkit-print-color-adjust:exact;color-adjust:exact;}}.css-1e6y48t-MuiButtonBase-root-MuiButton-root:hover{-webkit-text-decoration:none;text-decoration:none;background-color:rgba(25, 118, 210, 0.04);}@media (hover: none){.css-1e6y48t-MuiButtonBase-root-MuiButton-root:hover{background-color:transparent;}}.css-1e6y48t-MuiButtonBase-root-MuiButton-root.Mui-disabled{color:rgba(0, 0, 0, 0.26);}",
      }
    }
    data-emotion="css 1e6y48t-MuiButtonBase-root-MuiButton-root"
  />
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-i4bv87-MuiSvgIcon-root{-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;width:1em;height:1em;display:inline-block;fill:currentColor;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;-webkit-transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;font-size:1.5rem;}",
        }
      }
      data-emotion="css i4bv87-MuiSvgIcon-root"
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="FavoriteIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="m12 21.35-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
      />
    </svg>
  </button>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-j8ofbm-MuiDrawer-root{z-index:1200;}",
      }
    }
    data-emotion="css j8ofbm-MuiDrawer-root"
  />
</div>
`;
